runtime/nvqir/cutensornet/tn_simulation_state.cpp

```diff
@@ -152,6 +152,33 @@ TensorNetSimulationState::overlap(const cudaq::SimulationState &other) {
 
 std::complex<double>
 TensorNetSimulationState::getAmplitude(const std::vector<int> &basisState) {
+  if (getNumQubits() != basisState.size())
+    throw std::runtime_error(
+        fmt::format("[tensornet-state] getAmplitude with an invalid number "
+                    "of bits in the "
+                    "basis state: expected {}, provided {}.",
+                    getNumQubits(), basisState.size()));
```

Comment:
not this PR. I wonder if we can't remove most/all of our dependence on the `fmt` package by just building the strings with regular `std` functions.

Reply:
I think we can. It's just more convenient: our `Logger.h` makes `fmt` available, which is much nicer to use to construct these messages.